qiskit/chemistry/algorithms/minimum_eigen_solvers/oovqe.py

```diff
@@ -0,0 +1,604 @@
+            include_custom: When `expectation` parameter here is None setting this to ``True`` will
+                allow the factory to include the custom Aer pauli expectation.
+            max_evals_grouped: Max number of evaluations performed simultaneously. Signals the
+                given optimizer that more than one set of parameters can be supplied so that
+                potentially the expectation values can be computed in parallel. Typically this is
+                possible when a finite difference gradient is used by the optimizer such that
+                multiple points to compute the gradient can be passed and if computed in parallel
+                improve overall execution time.
+            aux_operators: Optional list of auxiliary operators to be evaluated with the eigenstate
+                of the minimum eigenvalue main result and their expectation values returned.
+                For instance in chemistry these can be dipole operators, total particle count
+                operators so we can get values for these at the ground state.
+            callback: a callback that can access the intermediate data during the optimization.
+                Four parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the evaluation count, the optimizer parameters for the
+                variational form, the evaluated mean and the evaluated standard deviation.
+            quantum_instance: :class:`~qiskit.aqua.QuantumInstance` that the algorithm is
+                 executed on.
+            orbital_rotation: instance of
+                :class:`~qiskit.chemistry.algorithms.minimum_eigen_solvers.OrbitalRotation` class
+                that creates the matrices that rotate the orbitals needed to produce the rotated
+                MO coefficients C as C = C0 * exp(-kappa).
+            core: instance of the :class:`~qiskit.chemistry.core.Hamiltonian` class to
+                 make new qubit operator after the orbital rotation using the data stored in
+                 qmolecule.
+            qmolecule: instance of the :class:`~qiskit.chemistry.QMolecule` class which has methods
+                needed to recompute one-/two-electron/dipole integrals after orbital rotation
+                (C = C0 * exp(-kappa)).
+            bounds: bounds for variational form and orbital rotation
+                 parameters given to a classical optimizer.
+            iterative_oo: when ``True`` optimize first the variational form and then the
+                orbitals, iteratively. Otherwise, the wavefunction ansatz and orbitals are
+                optimized simultaneously.
+            iterative_oo_iterations: number of iterations in the iterative procedure,
+                set larger to be sure to converge to the global minimum.
+
```

Comment:
Can you explain the interplay here with the optimizer max_iters or max fun evals parameterization. For normal VQE this limits the total number of evalutions before the optimizer will return. I think in the case of iterative_oo set False that this will be the same. But if iterative_oo is set True then the limit of total number of evaluations  will be iterative_oo_iterations * 2 * whatever the optimizer limit is.
Since this algo is oo would it be simpler to have the parameters just named `iterative` and, `iterations` or `num_iterations` - having just typed the iterative_oo_iterations it made me notice more. And I think this iterations here is always done right? Its not a max number like the optimizer where there is some early breakout when the minimum condition is determined to be found.

Reply:
Yes, I think iterative and num_iterations can be better names too. And yes exactly, it is not like optimizer that can stop if certain conditions are met. Iterative OOVQE just does num_iterations in any case. So lets rename as you suggest ...